src/shogun/features/Features.cpp

```diff
@@ -265,6 +265,13 @@ void CFeatures::unset_property(EFeatureProperty p)
 	properties &= (properties | p) ^ p;
 }
 
+CFeatures* CFeatures::view(const SGVector<index_t>& subset)
+{
+	auto feats_view = this->duplicate();
+	feats_view->add_subset(subset.clone());
```

Comment:
i wonder, do we want to clone the subset?
I think actually we shouldnt
And also I think that `add_subset` should accept a const vector

Reply:
but maybe you have a different opinion?

Reply:
agree, we don't need clone here